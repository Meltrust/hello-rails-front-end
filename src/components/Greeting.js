import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { v4 as uuidv4 } from 'uuid';
import { fetchGreetings } from '../redux/greetings/greetings';

const Greeting = () => {
  const dispatch = useDispatch();
  const greetingsData = useSelector((state) => state.greetingsReducer);

  useEffect(() => {
    dispatch(fetchGreetings());

    window.scrollTo(0, 0);
  }, []);

  if (greetingsData.loading) {
    return <h2 className="text-dark">Loading...</h2>;
  }
  if (greetingsData.error) {
    return <h2 className="text-dark">{greetingsData.error}</h2>;
  }

  return (
    <div className="d-flex mt-5 flex-column align-items-center" data-testid="greetingsList">

      <button type="button" key={uuidv4()} className="col-6 mb-5 btn btn-primary btn-lg" onClick={() => dispatch(fetchGreetings())}>Get greetings</button>

      <h2 className="col m-0 mt-2">{greetingsData.greetingsDisplay}</h2>
    </div>
  );
};

export default Greeting;
