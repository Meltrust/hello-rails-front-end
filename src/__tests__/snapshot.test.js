import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import store from '../redux/configureStore';
import Greeting from '../components/Greeting';

describe('Components snapshot testing', () => {
  test('Home page testing', () => {
    const homePage = renderer.create(
      <Provider store={store}>
        <Greeting />
      </Provider>,
    ).toJSON;
    expect(homePage).toMatchSnapshot();
  });
});
