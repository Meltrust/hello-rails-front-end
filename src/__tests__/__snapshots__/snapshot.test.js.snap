// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components snapshot testing Home page testing 1`] = `[Function]`;
